# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../sloccount/tail-head.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

This changes are needed for the new head and tail argument specification as
defined by POSIX.

  - Rene Rebe <rene@t2-project.org>

diff -ur sloccount-2.26/compute_sloc_lang sloccount-2.26-fixed/compute_sloc_lang
--- sloccount-2.26/compute_sloc_lang	2004-08-01 05:02:54.000000000 +0200
+++ sloccount-2.26-fixed/compute_sloc_lang	2005-07-20 10:02:13.000000000 +0200
@@ -53,7 +53,7 @@
      *) ${language}_count -f ${language}_list.dat > ${language}_outfile.dat
         ;;
    esac
-   tail -1 < ${language}_outfile.dat
+   tail -n 1 < ${language}_outfile.dat
 
   else
    rm -f ${language}_outfile.dat
diff -ur sloccount-2.26/redo_licenses sloccount-2.26-fixed/redo_licenses
--- sloccount-2.26/redo_licenses	2004-08-01 05:10:31.000000000 +0200
+++ sloccount-2.26-fixed/redo_licenses	2005-07-20 10:03:03.000000000 +0200
@@ -31,11 +31,11 @@
 cd $BUILD
 for builddir in *
 do
- specfile=`cat ${builddir}/ORIGINAL_SPEC_FILE | head -1`
+ specfile=`cat ${builddir}/ORIGINAL_SPEC_FILE | head -n 1`
  specfile=${SPECS}/$specfile
  echo "builddir=${builddir}, specfile=${specfile}"
  /root/extract_license "$builddir" "$specfile" > ${builddir}/PROGRAM_LICENSE
- license=`cat ${builddir}/PROGRAM_LICENSE | head -1`
+ license=`cat ${builddir}/PROGRAM_LICENSE | head -n 1`
  echo "   $license"
 
 done
